### PR TITLE
Rageeeee

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: config/formats.js [hunk 1]
```diff
@@ -1038,7 +1038,7 @@ let Formats = [
 		onSwitchInPriority: 100,
 		onSwitchIn(pokemon) {
 			let name = toID(pokemon.illusion ? pokemon.illusion.name : pokemon.name);
-			if (this.getTemplate(name).exists) {
+			if (this.getTemplate(name).exists || name === "rage") {
```
Comment (HoeenCoder) [inline config/formats.js]:
```suggestion
			if (this.getTemplate(name).exists || this.getEffect(name).exists) {
```

File: data/mods/ssb/moves.js [hunk 2]
```diff
@@ -3212,13 +3212,43 @@ let BattleMovedex = {
...
+			this.add('-anim', source, "Rage", target);
+		},
+		onHit(target, source) {
+			source.addVolatile('rage', source);
+			source.addVolatile('endure', source);
+			target.addVolatile('hitsmorethanonce', source);
+		},
+		secondary: null,
+		target: "normal",
+		type: "Normal",
```
Comment (HoeenCoder) [inline data/mods/ssb/moves.js]: Flying type

File: data/mods/ssb/random-teams.js [hunk 3]
```diff
@@ -790,6 +796,7 @@ class RandomStaffBrosTeams extends RandomTeams {
 			set.moves.push(ssbSet.signatureMove);
 			if (name === 'The Immortal' && set.item === 'Choice Scarf') set.moves[3] = 'Superpower';
 			if (name === 'irritated' && !set.moves.includes('Double Iron Bash')) set.moves[this.random(3)] = 'Double Iron Bash';
+			if (name === 'Rage') set.happiness = 0;
```
Comment (HoeenCoder) [inline data/mods/ssb/random-teams.js]: Instead of this, support a happiness value in the SSBSet type.

File: data/mods/ssb/statuses.js [hunk 4]
```diff
@@ -1595,6 +1607,22 @@ let BattleStatuses = {
 			this.add('-end', pokemon, 'Attract', '[silent]');
 		},
 	},
+	// Custom effect for Rage's multihit
+	hitsmorethanonce: {
```
Comment (HoeenCoder) [inline data/mods/ssb/statuses.js]: Maybe call it `enrageeeeed` because I feel that fits better.

Comment (ghost) [reply data/mods/ssb/statuses.js]: ._.

File: data/mods/ssb/random-teams.js (hunk 3 above)
Comment (ghost) [reply data/mods/ssb/random-teams.js]: I did try doing that before but I couldn't get it to work

Comment (HoeenCoder) [reply data/mods/ssb/random-teams.js]: `@property {number=} happiness`

File: data/mods/ssb/statuses.js (hunk 4 above)
Comment (HoeenCoder) [reply data/mods/ssb/statuses.js]: :3

**Pull request merged**